cli/flox/src/commands/environment.rs

```diff
@@ -1218,24 +1376,38 @@ impl Update {
     pub async fn handle(self, flox: Flox) -> Result<()> {
         subcommand_metric!("update");
 
-        let ((old_lockfile, new_lockfile), global, description) = match self.environment_or_global {
+        let (old_lockfile, new_lockfile, global, description) = match self.environment_or_global {
             EnvironmentOrGlobalSelect::Environment(ref environment_select) => {
                 let concrete_environment =
                     environment_select.detect_concrete_environment(&flox, "update")?;
 
                 let description = Some(environment_description(&concrete_environment)?);
-
+                let (old_manifest, new_manifest) =
+                    self.update_manifest(flox, concrete_environment)?;
                 (
-                    self.update_manifest(flox, concrete_environment)?,
+                    old_manifest
+                        .map(TypedLockedManifest::try_from)
+                        .transpose()?,
+                    TypedLockedManifest::try_from(new_manifest)?,
                     false,
                     description,
                 )
             },
-            EnvironmentOrGlobalSelect::Global => (self.update_global_manifest(flox)?, true, None),
+            EnvironmentOrGlobalSelect::Global => {
+                let (old_manifest, new_manifest) = self.update_global_manifest(flox)?;
```

Comment:
nit: I don't like calling the lockfile the manifest. I also prefer lockfile to locked manifest.